tests/test_ingestion.py

```diff
@@ -149,46 +165,67 @@ def setUpDefectRegistry(self):
         self._registerTask = patcherRegister.start()
         self._registerTask.config = self._task.config.defectIngester.register
         self.addCleanup(patcherRegister.stop)
-        # the mocked entry point of the Registry context manager, needed for querying the registry
-        self._registryHandle = self._registerTask.openRegistry().__enter__()
+
+    def assertRegistryCalls(self, registryMock, expectedData, calib):
+        kwargs = {'create': False, 'dryrun': False}
+        for datum in expectedData:
+            # TODO: find a way to avoid having to know exact data ID expansion
+            if calib:
+                dataId = {'calibDate': datum['date'], 'filter': datum['filter']}
+                kwargs['table'] = datum['type']
+            else:
+                dataId = {'visit': datum['visit'], 'expTime': datum['exptime'], 'filter': datum['filter']}
+            # TODO: I don't think we actually care about the keywords -- especially since they're defaults
+            registryMock.addRow.assert_any_call(registryMock.openRegistry().__enter__(), dataId,
+                                                **kwargs)
+
+        self.assertEqual(registryMock.addRow.call_count, len(expectedData))
 
     def testDataIngest(self):
-        """Test that ingesting a science image adds it to a repository.
+        """Test that ingesting a science image using a low-level method adds it to a repository.
         """
         self.setUpRawRegistry()
-        testDir = os.path.join(IngestionTestSuite.testData, 'raw')
-        files = [os.path.join(testDir, datum['file']) for datum in IngestionTestSuite.rawData]
+        files = [os.path.join(self._dataset.rawLocation, datum['file'])
+                 for datum in IngestionTestSuite.rawData]
         self._task._doIngestRaws(self._repo, files, [])
 
-        for datum in IngestionTestSuite.rawData:
-            # TODO: find a way to avoid having to know exact data ID expansion
-            dataId = {'visit': datum['visit'], 'expTime': datum['exptime'], 'filter': datum['filter']}
-            # TODO: I don't think we actually care about the keywords -- especially since they're defaults
-            self._registerTask.addRow.assert_any_call(self._registryHandle, dataId,
-                                                      create=False, dryrun=False)
-        self.assertEqual(self._registerTask.addRow.call_count, len(IngestionTestSuite.rawData))
+        self.assertRegistryCalls(self._registerTask, IngestionTestSuite.rawData, calib=False)
+
+    def testDataIngestDriver(self):
+        """Test that ingesting science images using a high-level method adds them to a repository.
```

Comment:
It tests dataset-management and file-finding logic that the `testDataIngest` test bypasses. In fact, I discovered the bug fixed in 461eea3984bb5f6459084953e35118e4a788e7b6 because of this test.

Reply:
Ah! Might be worth expanding the docstring(s) to explain that.